### ISSUE TITLE
Collecting matplotlib>=3.0

### ISSUE THREAD

**Issue by ludwigschreier** [uber/ludwig#347]
Problem with matplotlib at installation.

~/dev/git/ludwig$ sudo pip install -r requirements.txt 

```
(venv) ludwig1@ubuntu:~/dev/git/ludwig$ sudo pip install -r requirements.txt 
The directory '/home/ludwig1/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/ludwig1/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting Cython>=0.25 (from -r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/16/7d/30ab222995d7dd551dee875c9f8fecf94b3c16a24f1c34d9a93031b19209/Cython-0.29.7-cp27-cp27mu-manylinux1_x86_64.whl (2.0MB)
    100% |████████████████████████████████| 2.0MB 521kB/s 
Collecting h5py>=2.6 (from -r requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/53/08/27e4e9a369321862ffdce80ff1770553e9daec65d98befb2e14e7478b698/h5py-2.9.0-cp27-cp27mu-manylinux1_x86_64.whl (2.8MB)
    100% |████████████████████████████████| 2.8MB 419kB/s 
Collecting matplotlib>=3.0 (from -r requirements.txt (line 3))
  Could not find a version that satisfies the requirement matplotlib>=3.0 (from -r requirements.txt (line 3)) (from versions: 0.86, 0.86.1, 0.86.2, 0.91.0, 0.91.1, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1rc1, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.1.0rc1, 2.1.0, 2.1.1, 2.1.2, 2.2.0rc1, 2.2.0, 2.2.2, 2.2.3, 2.2.4)
No matching distribution found for matplotlib>=3.0 (from -r requirements.txt (line 3))

```

This is a VM running 64 Bit Ubuntu.

```
(venv) ludwig1@ubuntu:~/dev/git/ludwig$ uname -a
Linux ubuntu 4.18.0-18-generic #19~18.04.1-Ubuntu SMP Fri Apr 5 10:22:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

```
(venv) ludwig1@ubuntu:~/dev/git/ludwig$ pip -V
pip 19.1.1 from /home/ludwig1/dev/git/ludwig/venv/lib/python3.6/site-packages/pip (python 3.6)

```

```
ludwig1@ubuntu:~/dev/git/ludwig/ludwig$ pip install ludwig
Collecting ludwig
  Cache entry deserialization failed, entry ignored
  Could not find a version that satisfies the requirement ludwig (from versions: )
No matching distribution found for ludwig
```
Doesn't work this way either. Anaconda was used and Python "downgraded" using "conda install python=3.6.5" command.

Regards

**Comment by w4nderlust** [comment]
Can't reproduce unfortunately, on my machines matplotlib installs just fine.
Try not downgrading python and using a virtual environment as suggested in getting started guide, never had problems with that. Let me know if it solves your problem.

**Comment by ludwigschreier** [comment]
Well, I can try again but in principle, neither installation paths as described (pip or manual installation etc.) did work at first try.

**Comment by w4nderlust** [comment]
I'm quoting directly from the getting started guide:
```
sudo apt install python3  # on ubuntu
brew install python3      # on mac

You may want to use a virtual environment to maintain an isolated Python environment.
In order to install Ludwig just run:

pip install ludwig

or install it by building the source code from the repository:

git clone git@github.com:uber/ludwig.git
cd ludwig
virtualenv -p python3 venv
source venv/bin/activate
pip install -r requirements.txt
python setup.py install
```
We suggest to use a virtual environment, have you done so? Did you follow exactly those instructions?
I'm not asking you to redo what you have already done, I'm suggesting to follow the instructions given in the guide, without Anaconda and without any downgraded python as there's nothing in the documentations that suggests so and using a virtual environment with python3.

If following exactly those instructions does not lead you to a working environment then there could be something wrong with Ludwig's installation procedure, but most likely there's something wrong with your environment and setup.

**Comment by ludwigschreier** [comment]
Hey,
yes, of course I would start and have started with the getting started guide as it looks straight forward and inviting (despite that I found that examples such as *.csv's and *.yaml's are not provided as files, but part of documentation only? - separate question)

Python installation:

```
ludwig1@ubuntu:~/dev/git/ludwig/ludwig$ sudo apt install python3
[sudo] password for ludwig1: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python3 is already the newest version (3.6.7-1~18.04).
0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
```

```
ludwig1@ubuntu:~/dev$ pip install ludwig
Collecting ludwig
  Could not find a version that satisfies the requirement ludwig (from versions: )
No matching distribution found for ludwig

```

There was a related bug here, but that was about TF: https://github.com/uber/ludwig/issues/102

Then, running virtual env:

```
(venv) ludwig1@ubuntu:~/dev/git/ludwig$ pip install -r requirements.txt 
Collecting Cython>=0.25 (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/b2/20/46a78072ecd4fda072c3791a257b03af99b64673671663067d18bc4935ec/Cython-0.29.7-cp36-cp36m-manylinux1_x86_64.whl
Collecting h5py>=2.6 (from -r requirements.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/30/99/d7d4fbf2d02bb30fb76179911a250074b55b852d34e98dd452a9f394ac06/h5py-2.9.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting matplotlib>=3.0 (from -r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/da/83/d989ee20c78117c737ab40e0318ea221f1aed4e3f5a40b4f93541b369b93/matplotlib-3.1.0-cp36-cp36m-manylinux1_x86_64.whl (13.1MB)
     |████████████████████████████████| 13.1MB 1.3MB/s 
Collecting numpy>=1.15 (from -r requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/c1/e2/4db8df8f6cddc98e7d7c537245ef2f4e41a1ed17bf0c3177ab3cc6beac7f/numpy-1.16.3-cp36-cp36m-manylinux1_x86_64.whl
Collecting pandas>=0.19 (from -r requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/19/74/e50234bc82c553fecdbd566d8650801e3fe2d6d8c8d940638e3d8a7c5522/pandas-0.24.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting scipy>=0.18 (from -r requirements.txt (line 6))
  Using cached https://files.pythonhosted.org/packages/72/4c/5f81e7264b0a7a8bd570810f48cd346ba36faedbd2ba255c873ad556de76/scipy-1.3.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting scikit-image==0.14.2 (from -r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/24/06/d560630eb9e36d90d69fe57d9ff762d8f501664ce478b8a0ae132b3c3008/scikit_image-0.14.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting scikit-learn (from -r requirements.txt (line 8))
  Using cached https://files.pythonhosted.org/packages/90/c7/401c231c445fb6fad135e92197da9c3e77983de169ff1887cc18af94498d/scikit_learn-0.21.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting seaborn>=0.7 (from -r requirements.txt (line 9))
  Using cached https://files.pythonhosted.org/packages/a8/76/220ba4420459d9c4c9c9587c6ce607bf56c25b3d3d2de62056efe482dadc/seaborn-0.9.0-py3-none-any.whl
Collecting spacy>=2.1 (from -r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/a1/5b/0fab3fa533229436533fb504bb62f4cf7ea29541a487a9d1a0749876fc23/spacy-2.1.4-cp36-cp36m-manylinux1_x86_64.whl
Collecting tqdm (from -r requirements.txt (line 11))
  Using cached https://files.pythonhosted.org/packages/45/af/685bf3ce889ea191f3b916557f5677cc95a5e87b2fa120d74b5dd6d049d0/tqdm-4.32.1-py2.py3-none-any.whl
Collecting tabulate>=0.7 (from -r requirements.txt (line 12))
Collecting tensorflow==1.13.1 (from -r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/77/63/a9fa76de8dffe7455304c4ed635be4aa9c0bacef6e0633d87d5f54530c5c/tensorflow-1.13.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting PyYAML>=3.12 (from -r requirements.txt (line 14))
Collecting pytest (from -r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/56/53/0ae37ab12c457945ae0152c6571d6d40eecccddf25f71fe328f9aefe90ca/pytest-4.5.0-py2.py3-none-any.whl
Collecting six (from h5py>=2.6->-r requirements.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting python-dateutil>=2.1 (from matplotlib>=3.0->-r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib>=3.0->-r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/f8/a1/5742b56282449b1c0968197f63eae486eca2c35dcd334bab75ad524e0de1/kiwisolver-1.1.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib>=3.0->-r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/dd/d9/3ec19e966301a6e25769976999bd7bbe552016f0d32b577dc9d63d2e0c49/pyparsing-2.4.0-py2.py3-none-any.whl
Collecting cycler>=0.10 (from matplotlib>=3.0->-r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/f7/d2/e07d3ebb2bd7af696440ce7e754c59dd546ffe1bbe732c8ab68b9c834e61/cycler-0.10.0-py2.py3-none-any.whl
Collecting pytz>=2011k (from pandas>=0.19->-r requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/3d/73/fe30c2daaaa0713420d0382b16fbb761409f532c56bdcc514bf7b6262bb6/pytz-2019.1-py2.py3-none-any.whl
Collecting pillow>=4.3.0 (from scikit-image==0.14.2->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/d2/c2/f84b1e57416755e967236468dcfb0fad7fd911f707185efc4ba8834a1a94/Pillow-6.0.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting networkx>=1.8 (from scikit-image==0.14.2->-r requirements.txt (line 7))
Collecting cloudpickle>=0.2.1 (from scikit-image==0.14.2->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/24/fb/4f92f8c0f40a0d728b4f3d5ec5ff84353e705d8ff5e3e447620ea98b06bd/cloudpickle-1.1.1-py2.py3-none-any.whl
Collecting PyWavelets>=0.4.0 (from scikit-image==0.14.2->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/4e/cd/528dba0b474b08f6f9a3a5e1b4bb23d8e33ed5d9f0e321cc967c2607df05/PyWavelets-1.0.3-cp36-cp36m-manylinux1_x86_64.whl
Collecting dask[array]>=1.0.0 (from scikit-image==0.14.2->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/58/80/6059587dcf80821fbccdc1f06b6d3fd29b64c9b38fbf1ea42889460de3de/dask-1.2.2-py2.py3-none-any.whl
Collecting joblib>=0.11 (from scikit-learn->-r requirements.txt (line 8))
  Using cached https://files.pythonhosted.org/packages/cd/c1/50a758e8247561e58cb87305b1e90b171b8c767b15b12a1734001f41d356/joblib-0.13.2-py2.py3-none-any.whl
Collecting jsonschema<3.1.0,>=2.6.0 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/aa/69/df679dfbdd051568b53c38ec8152a3ab6bc533434fc7ed11ab034bf5e82f/jsonschema-3.0.1-py2.py3-none-any.whl
Collecting requests<3.0.0,>=2.13.0 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting thinc<7.1.0,>=7.0.2 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/a9/f1/3df317939a07b2fc81be1a92ac10bf836a1d87b4016346b25f8b63dee321/thinc-7.0.4-cp36-cp36m-manylinux1_x86_64.whl
Collecting wasabi<1.1.0,>=0.2.0 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/f4/c1/d76ccdd12c716be79162d934fe7de4ac8a318b9302864716dde940641a79/wasabi-0.2.2-py3-none-any.whl
Collecting blis<0.3.0,>=0.2.2 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/34/46/b1d0bb71d308e820ed30316c5f0a017cb5ef5f4324bcbc7da3cf9d3b075c/blis-0.2.4-cp36-cp36m-manylinux1_x86_64.whl
Collecting srsly<1.1.0,>=0.0.5 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/6b/97/47753e3393aa4b18de9f942fac26f18879d1ae950243a556888f389d1398/srsly-0.0.5-cp36-cp36m-manylinux1_x86_64.whl
Collecting cymem<2.1.0,>=2.0.2 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/3d/61/9b0520c28eb199a4b1ca667d96dd625bba003c14c75230195f9691975f85/cymem-2.0.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting plac<1.0.0,>=0.9.6 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/9e/9b/62c60d2f5bc135d2aa1d8c8a86aaf84edb719a59c7f11a4316259e61a298/plac-0.9.6-py2.py3-none-any.whl
Collecting preshed<2.1.0,>=2.0.1 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/20/93/f222fb957764a283203525ef20e62008675fd0a14ffff8cc1b1490147c63/preshed-2.0.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting murmurhash<1.1.0,>=0.28.0 (from spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/a6/e6/63f160a4fdf0e875d16b28f972083606d8d54f56cd30cb8929f9a1ee700e/murmurhash-1.0.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting gast>=0.2.0 (from tensorflow==1.13.1->-r requirements.txt (line 13))
Collecting tensorflow-estimator<1.14.0rc0,>=1.13.0 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/bb/48/13f49fc3fa0fdf916aa1419013bb8f2ad09674c275b4046d5ee669a46873/tensorflow_estimator-1.13.0-py2.py3-none-any.whl
Collecting astor>=0.6.0 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/35/6b/11530768cac581a12952a2aad00e1526b89d242d0b9f59534ef6e6a1752f/astor-0.7.1-py2.py3-none-any.whl
Collecting termcolor>=1.1.0 (from tensorflow==1.13.1->-r requirements.txt (line 13))
Collecting keras-applications>=1.0.6 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/90/85/64c82949765cfb246bbdaf5aca2d55f400f792655927a017710a78445def/Keras_Applications-1.0.7-py2.py3-none-any.whl
Collecting absl-py>=0.1.6 (from tensorflow==1.13.1->-r requirements.txt (line 13))
Requirement already satisfied: wheel>=0.26 in ./venv/lib/python3.6/site-packages (from tensorflow==1.13.1->-r requirements.txt (line 13)) (0.33.4)
Collecting grpcio>=1.8.6 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/0a/9d/8bd5d0e516b196f59f1c4439b424b8d4fa62d492a4b531aae322d2d82a7b/grpcio-1.20.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting tensorboard<1.14.0,>=1.13.0 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/0f/39/bdd75b08a6fba41f098b6cb091b9e8c7a80e1b4d679a581a0ccd17b10373/tensorboard-1.13.1-py3-none-any.whl
Collecting protobuf>=3.6.1 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/5a/aa/a858df367b464f5e9452e1c538aa47754d467023850c00b000287750fa77/protobuf-3.7.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting keras-preprocessing>=1.0.5 (from tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/c0/bf/0315ef6a9fd3fc2346e85b0ff1f5f83ca17073f2c31ac719ab2e4da0d4a3/Keras_Preprocessing-1.0.9-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0; python_version > "2.7" (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/b3/73/64fb5922b745fc1daee8a2880d907d2a70d9c7bb71eea86fcb9445daab5e/more_itertools-7.0.0-py3-none-any.whl
Collecting attrs>=17.4.0 (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/23/96/d828354fa2dbdf216eaa7b7de0db692f12c234f7ef888cc14980ef40d1d2/attrs-19.1.0-py2.py3-none-any.whl
Collecting wcwidth (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
Collecting pluggy!=0.10,<1.0,>=0.9 (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/cc/e4/2003a5e4e445424602c8f06a436eeed9e78bcae4947128f64741c66c7216/pluggy-0.11.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./venv/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 15)) (41.0.1)
Collecting py>=1.5.0 (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/76/bc/394ad449851729244a97857ee14d7cba61ddb268dce3db538ba2f2ba1f0f/py-1.8.0-py2.py3-none-any.whl
Collecting atomicwrites>=1.0 (from pytest->-r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/52/90/6155aa926f43f2b2a22b01be7241be3bfd1ceaf7d0b3267213e8127d41f4/atomicwrites-1.3.0-py2.py3-none-any.whl
Collecting decorator>=4.3.0 (from networkx>=1.8->scikit-image==0.14.2->-r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Collecting toolz>=0.7.3; extra == "array" (from dask[array]>=1.0.0->scikit-image==0.14.2->-r requirements.txt (line 7))
Collecting pyrsistent>=0.14.0 (from jsonschema<3.1.0,>=2.6.0->spacy>=2.1->-r requirements.txt (line 10))
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.13.0->spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/60/75/f692a584e85b7eaba0e03827b3d51f45f571c2e793dd731e598828d380aa/certifi-2019.3.9-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5 (from requests<3.0.0,>=2.13.0->spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests<3.0.0,>=2.13.0->spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests<3.0.0,>=2.13.0->spacy>=2.1->-r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/39/ec/d93dfc69617a028915df914339ef66936ea976ef24fa62940fd86ba0326e/urllib3-1.25.2-py2.py3-none-any.whl
Collecting mock>=2.0.0 (from tensorflow-estimator<1.14.0rc0,>=1.13.0->tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/05/d2/f94e68be6b17f46d2c353564da56e6fb89ef09faeeff3313a046cb810ca9/mock-3.0.5-py2.py3-none-any.whl
Collecting markdown>=2.6.8 (from tensorboard<1.14.0,>=1.13.0->tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/f5/e4/d8c18f2555add57ff21bf25af36d827145896a07607486cc79a2aea641af/Markdown-3.1-py2.py3-none-any.whl
Collecting werkzeug>=0.11.15 (from tensorboard<1.14.0,>=1.13.0->tensorflow==1.13.1->-r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/9f/57/92a497e38161ce40606c27a86759c6b92dd34fcdb33f64171ec559257c02/Werkzeug-0.15.4-py2.py3-none-any.whl
Installing collected packages: Cython, six, numpy, h5py, python-dateutil, kiwisolver, pyparsing, cycler, matplotlib, pytz, pandas, scipy, pillow, decorator, networkx, cloudpickle, PyWavelets, toolz, dask, scikit-image, joblib, scikit-learn, seaborn, pyrsistent, attrs, jsonschema, certifi, idna, chardet, urllib3, requests, blis, tqdm, cymem, preshed, srsly, plac, murmurhash, wasabi, thinc, spacy, tabulate, gast, absl-py, mock, tensorflow-estimator, astor, termcolor, keras-applications, grpcio, markdown, protobuf, werkzeug, tensorboard, keras-preprocessing, tensorflow, PyYAML, more-itertools, wcwidth, pluggy, py, atomicwrites, pytest
ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/home/ludwig1/dev/git/ludwig/venv/lib/python3.6/site-packages/cython.py'
Consider using the `--user` option or check the permissions.
```

and as _root_:

```
(venv) ludwig1@ubuntu:~/dev/git/ludwig$ sudo pip install -r requirements.txt 
The directory '/home/ludwig1/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/ludwig1/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting Cython>=0.25 (from -r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/16/7d/30ab222995d7dd551dee875c9f8fecf94b3c16a24f1c34d9a93031b19209/Cython-0.29.7-cp27-cp27mu-manylinux1_x86_64.whl (2.0MB)
    100% |████████████████████████████████| 2.0MB 436kB/s 
Collecting h5py>=2.6 (from -r requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/53/08/27e4e9a369321862ffdce80ff1770553e9daec65d98befb2e14e7478b698/h5py-2.9.0-cp27-cp27mu-manylinux1_x86_64.whl (2.8MB)
    100% |████████████████████████████████| 2.8MB 502kB/s 
Collecting matplotlib>=3.0 (from -r requirements.txt (line 3))
  Could not find a version that satisfies the requirement matplotlib>=3.0 (from -r requirements.txt (line 3)) (from versions: 0.86, 0.86.1, 0.86.2, 0.91.0, 0.91.1, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1rc1, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.1.0rc1, 2.1.0, 2.1.1, 2.1.2, 2.2.0rc1, 2.2.0, 2.2.2, 2.2.3, 2.2.4)
No matching distribution found for matplotlib>=3.0 (from -r requirements.txt (line 3))
```

**Comment by msaisumanth** [comment]
@ludwigschreier Have you tried installing Ludwig directly from the source code? That won't solve the matplotlib issue but you shouldn't have issues installing Ludwig.

**Comment by w4nderlust** [comment]
@ludwigschreier  i see no sign from your logs that you created a virtual environment and you are inside it, as I suggested to do previously.

**Comment by ludwigschreier** [comment]
Hey, the **(venv)** indicates the virtual environment doesn't it?

I started in another VM box, from scratch, using Ubuntu.


```
ludwig1@ubuntu:~$ pip install ludwig
Collecting ludwig
  Could not find a version that satisfies the requirement ludwig (from versions: )
No matching distribution found for ludwig
```

If I follow the introduction, I should just use pip to install, but that doesn't work. So what next, I will try the manual installation using virtual env.

**Comment by ludwigschreier** [comment]
OK, here is some good news.
After installation using pip3 instead of pip (and it was successful compared to the other VM still stuck at matplotlib), I can reproduce this output:

```
ludwig1@ubuntu:~$ python3 -m ludwig.train -h

WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

usage: ludwig train [options]

This script trains a model.

```

However, command 'ludwig' is not available in terminal.

**Comment by w4nderlust** [comment]
> Hey, the **(venv)** indicates the virtual environment doesn't it?

yes but if you don't show me that the venv you created is a python 3 venv by showing either how you created it or what version is the python inside it i can't be sure you are doing things correctly.

> OK, here is some good news.
> After installation using pip3 instead of pip (and it was successful compared to the other VM still stuck at matplotlib), I can reproduce this output:
> 
> ```
> ludwig1@ubuntu:~$ python3 -m ludwig.train -h
> ```
Now it looks again that you outside the virtual environment. If you are withing a virtual environment initialized with python3 as the interpreter you don't need to use the python3 command but just the python command. Also I don't know in which directory you are when you are running this, if you actually have it installed or if you are running it from source, from the directory you pulled.
As this is getting pretty messy, I suggest you to read some guide on virtual environments, for instance [this one](https://docs.python-guide.org/dev/virtualenvs/) that will probably clarify all your doubts. Also I'm closing this issue as it's likely not a Ludwig issue but an issue with your configuration. You can keep on writing here if you want and I'll keep trying to help.
